### DISCUSSION TITLE
Realistic modified ride stats, especially for rollercoasters

### DISCUSSION THREAD

**Discussion by ThiaJay** [OpenRCT2/OpenRCT2#18166] [Feature requests]
I'd like to see more realistic statistics for rollercoasters in particular, so we're not penalised for track length, number of drops and more reasonable allowances for number of inversions.

**Comment by astraylife1** [comment]
I agree. It is frustrating to try and build coasters with more than 10 inversions or coasters with 15 or more drops. No matter how slow the coaster goes over the hills, the intensity will be too high.

**Reply by ThiaJay** [reply]
I know some people like to play the game "pure", but with the introduction of OpenRCT, that's less of a concern given that it's primarily used to extend functionality of the game.

If anyone felt this was a step too far, there could be different profiles ranging from original to realistic. For those who wanted a laugh, there could be profiles for higher and lower gravity too.

**Comment by 733737** [comment]
what coasters in real life have more than 10 inversions?

**Reply by astraylife1** [reply]
Smiler at Alton Towers

**Reply by ThiaJay** [reply]
I love Smiler.

**Comment by ThiaJay** [comment]
Quick search: https://coastercritic.com/2014/02/roller-coasters-with-most-loops/

That's from 2014, there may be more now.

**Reply by astraylife1** [reply]
https://rcdb.com/rhr.htm?m=4

**Reply by ThiaJay** [reply]
Nice link

**Reply by ocalhoun6** [reply]
I think the fact that there's relatively few real-life coasters with that many inversions kind of implies a bit of realism in RCT2's stat calculations of them.

All of these record holders *are* extremely intense rides, and probably do scare away a significant portion of guests. And while it may make sense to make that sacrifice for the occasional flagship coaster that brings people to your park because it's so impressive ... *most* coasters in the world are built with a far more reasonable number of inversions.

An RCT recreation of Smiler probably *should* have around 10 intensity, if not a bit higher, since that's about as intense as real-life coasters get. Has anybody built one that's at least close to the original? What are the stats on it?

**Reply by astraylife1** [reply]
I would disagree. The RCT games are from two decades ago. The first ten-inversion coaster opened the year RCT2 was released. The game was designed at a time when only one giga coaster existed. Today, most new coasters are much more intense and I think the stat requirements should be adjusted to reflect that. 

Having the intensity increase for every drop on a coaster regardless of speed isn’t realistic. Steel Vengeance has an insane amount of drops and is an extremely popular ride. In RCT2, having an intensity over 10 would mean less peeps queue up which doesn’t reflect the realism of how popular it actually should be. The game was just not designed with these things in mind since these coasters in real life did not exist yet. 

I’m all for updating the stats so that we can create more realistic rides. 

I created a family-style mine train coaster that was not fast at all. But the number of drops at 15 gave it a 9.89 intensity. Not realistic at all. 

The game will count a small drop out of the station before the lift as a drop that adds to that intensity calc. Just not realistic to me. 

If we are going to add Hybrid and Single Rail coasters with their own stat calculations to increase the fun of the game and realism, why not update existing stat calcs for certain rides?

**Reply by 733737** [reply]
I'm all for removing the stat penalty on number of drops, but that's a limited scope debate unrelated to inversions which definitely should contribute to intensity

**Reply by ThiaJay** [reply]
> I would disagree. The RCT games are from two decades ago. The first ten-inversion coaster opened the year RCT2 was released. The game was designed at a time when only one giga coaster existed. Today, most new coasters are much more intense and I think the stat requirements should be adjusted to reflect that.
> 
> Having the intensity increase for every drop on a coaster regardless of speed isn’t realistic. Steel Vengeance has an insane amount of drops and is an extremely popular ride. In RCT2, having an intensity over 10 would mean less peeps queue up which doesn’t reflect the realism of how popular it actually should be. The game was just not designed with these things in mind since these coasters in real life did not exist yet.
> 
> I’m all for updating the stats so that we can create more realistic rides.
> 
> I created a family-style mine train coaster that was not fast at all. But the number of drops at 15 gave it a 9.89 intensity. Not realistic at all.
> 
> The game will count a small drop out of the station before the lift as a drop that adds to that intensity calc. Just not realistic to me.
> 
> If we are going to add Hybrid and Single Rail coasters with their own stat calculations to increase the fun of the game and realism, why not update existing stat calcs for certain rides?

Well put, I agree. The roughest rollercoasters I have ever been on were wooden, particularly those at Blackpool Pleasure Beach and Gwazi at Busch Gardens Tampa Bay (pre-hybrid). Those were so rough, yet I doubt the roughness of wooden rollercoasters is really factored into RCT 2. Whilst inversions do contribute to intensity, so does the roughness of a ride.

I went on Wicker Man at Alton Towers this year and I was impressed by how smooth it was compared to these other wooden rollercoasters, although I don't know how that could be factored into the game.

**Reply by X123M3-256** [reply]
> If we are going to add Hybrid and Single Rail coasters with their own stat calculations to increase the fun of the game and realism, why not update existing stat calcs for certain rides?

The hybrid and single rail coasters don't have a better stat calculation, it was implemented the same way as all the others for consistency.

You can't fix the problem just by tweaking the numbers. I tried to choose the parameters for the hybrid and single rail coaster to make the stats behave somewhat reasonably, but it was essentially impossible. There was very little difference between good layouts and terrible layouts, if they met the minimum requirements and don't hit any stat penalties.

I don't really think that OpenRCT2 should modify the stat penalties, because I think there should be two modes - the original one that works exactly as it did in vanilla for people who want to play the game as it was originally, and a realistic one that works completely differently and doesn't have hard stat penalties because they make no sense.

**Reply by astraylife1** [reply]
It would be nice to get rid of the stat penalties as it ruins the ratings for shoe strings and other new creations. It also ruins the fun creating a beautiful coaster or ride and then having horrible stats because the drop wasn’t high enough or the lateral Gs aren’t enough, etc.

I’ve always viewed OpenRCT2 as the future and making a more modern version of a classic game. Maybe this is an unpopular opinion but for those that want to play the vanilla game, they can purchase RCT Classic.

**Reply by ThiaJay** [reply]
I agree with you both.

**Reply by Generalcamo** [reply]
OpenRCT2 is supposed to appeal to both the sandboxer, and scenario players (as evidenced by the replacement Scenario project). Removing stat penalties would reward things like making a flat circle for a roller coaster, which would be detrimental for scenario players. 

Ideally, sandbox players would have options in either cheats or plugins, like having guests ignore EIN ratings or being able to override EIN settings on rides.

**Reply by ThiaJay** [reply]
> OpenRCT2 is supposed to appeal to both the sandboxer, and scenario players (as evidenced by the replacement Scenario project). Removing stat penalties would reward things like making a flat circle for a roller coaster, which would be detrimental for scenario players.
> 
> Ideally, sandbox players would have options in either cheats or plugins, like having guests ignore EIN ratings or being able to override EIN settings on rides.

If not default, at least have the ability to switch profiles between vanilla and realistic.

**Comment by ocalhoun6** [comment]
To help put this issue to rest, allow me to introduce my new test subject: Shitty Smiler.


![Hydro Hills 2022-10-05 12-45-55](https://user-images.githubusercontent.com/26573303/194149361-38cd36ed-48ad-4ad9-8d57-7eb4aea5585d.png)

11 inversions, and only 8.87 intensity.

So it's definitely *possible* to make rides with large numbers of inversions that still have reasonable stats.

I think what's really driving your stats over the top is the number of drops ... and I *do* think that maybe there's a conversation to be had about lessening the intensity penalty for number of drops, especially if a lot of the drops are small. Maybe it should be calculated based more on 'total height of all drops' instead.

**Reply by astraylife1** [reply]
Or having a cap where after 10 drops, there won’t be any additional intensity added for drops. Or only counting drops of a certain height.

**Reply by X123M3-256** [reply]
> So it's definitely possible to make rides with large numbers of inversions that still have reasonable stats.

Yeah, but this is a stupid layout that *shouldn't* get reasonable stats, so it really just highlights the problem even more.

**Reply by 733737** [reply]
I don't think the stats are a problem. If we change stats, we want to leave them close to vanilla. I'm ok with changing stat calculation so it's more cpu intensive and goes through more checks, but please don't rebalance the entire stat calculation without adding any complexity to it or at least changing variables such as drop limit

**Reply by astraylife1** [reply]
If people want vanilla, why don’t they just play vanilla? I don’t understand wanting OpenRCT2 to have all these new features but then also not wanting anything to change…

**Reply by ocalhoun6** [reply]
>If people want vanilla, why don’t they just play vanilla?

- Monitor resolution & scaling
- Linux compatibility
- QoL improvments
- Multiplayer
- etc, etc, etc...

No, we shouldn't make OpenRCT2 different just for the sake of being different. A lot of people out there *do* enjoy Open because it's a better version of RCT2, not because they want something entirely different.

**Reply by astraylife1** [reply]
It already is different than vanilla. Example, miniature railroad has at-grade crossings over paths. If you want to play a scenario in vanilla manner (railroad cannot be built through paths), this small mundane change to one ride has basically added a cheat to the game that cannot be turned off. Vanilla gameplay is impossible with all the small and large changes that have occurred to the game. Features such as my example have changed gameplay by improving upon the original. This improvement was done for realism and added value to players. Removing stat limitations and/or updating the calculations can be argued as just another change that improves gameplay for added realism and value.

**Reply by 733737** [reply]
you can argue all day, but that's not the same as winning a debate with new elements designs

**Reply by astraylife1** [reply]
To each their own. As for the subject of this idea discussion, the stats should change to be more realistic to current real world coaster designs. If people want vanilla, play vanilla. OpenRCT2 is not vanilla RCT and never will be by design.

**Reply by ThiaJay** [reply]
Rather than an argument for/against, I think this should be more of a discussion of whether this is hardcoded into OpenRCT or added as an option between original and realistic.

**Reply by X123M3-256** [reply]
> Rather than an argument for/against, I think this should be more of a discussion of whether this is hardcoded into OpenRCT or added as an option between original and realistic.

There's no way it would be hardcoded and not optional. If it's implemented as an official feature, it's going to be an option. They aren't going to break from vanilla in such a major way, a realistic stat calculation would make some existing designs get far worse stats than they currently do and people won't like that.

Remember when they corrected the loop clearance boxes? That was fixing what was clearly a bug in vanilla, but because it broke existing designs, people complained and it was reverted.

**Reply by 733737** [reply]
i didn't realize that got reverted

**Reply by ThiaJay** [reply]
> > Rather than an argument for/against, I think this should be more of a discussion of whether this is hardcoded into OpenRCT or added as an option between original and realistic.
> 
> There's no way it would be hardcoded and not optional. If it's implemented as an official feature, it's going to be an option. They aren't going to break from vanilla in such a major way, a realistic stat calculation would make some existing designs get far worse stats than they currently do and people won't like that.
> 
> Remember when they corrected the loop clearance boxes? That was fixing what was clearly a bug in vanilla, but because it broke existing designs, people complained and it was reverted.

That's fine with me.

Loop clearance boxes?

**Comment by ThiaJay** [comment]
The fact that track intensity is low on that is proof to me it needs looking into. Inversions vary in size and style, that looks more like the Euthanasia Coaster.

**Reply by 733737** [reply]
8.87 is not low intensity, any experienced rct player would know that intensity over 7 is not a popular ride with nearly half of the guests

**Reply by ThiaJay** [reply]
If you want to talk about realistic ride popularity, Smiler is a ride that makes me seriously dizzy, yet I still went on it three times and the queues were really long. So, we have established that high intensity does not always equate to low interest.

**Reply by ocalhoun6** [reply]
If you were to build Smiler in RCT2 using the stats as currently calculated, what intensity do you think it would have?
